### DISCUSSION TITLE
BRouter tracktype=grade1

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#180] [Q&A] [answered]
Danke an alle. Das installieren von Brouter und erste Versuche haben soweit geklappt.
Als Fan von kleinen Routen kann ich folgende Route nicht planen welche ich schon gefahren bin und welche regulär zu befahren ist. In der alten kurviger app ist das problemlos möglich.Wo liegt mein Fehler ?
https://kurv.gr/KzBMh

_Originally posted by @1maxi2 in https://github.com/devemux86/cruiser/discussions/175#discussioncomment-6010297_

**Comment by devemux86** [comment]
BRouter does not route across this bridge.
BRouter leitet nicht über diese Brücke.

- [Kurviger file](https://github.com/devemux86/cruiser/files/11574743/Gemeinde.Rossatz-Arnsdorf.-.Gemeinde.Weinzierl.am.Walde.kurviger.zip)
- [OpenStreetMap way](https://www.openstreetmap.org/way/508238652)
- [BRouter route](https://brouter.de/brouter-web/#map=18/48.38384/15.41317/standard&lonlats=15.413325,48.382624;15.4131,48.384626&profile=car-eco)

**Comment by devemux86** [comment]
@1maxi2

BRouter calculates the same route when we include the [highway=track tracktype=grade1](https://wiki.openstreetmap.org/wiki/Key:tracktype).
We also need to lower the vmax filter to e.g. 30 km/h.

Changed profile, line 66: [car-eco-cruiser.brf.zip](https://github.com/devemux86/cruiser/files/11575147/car-eco-cruiser.brf.zip)
I can include this change in the next Cruiser update.

---

BRouter berechnet die gleiche Route, wenn wir die [highway=track tracktype=grade1](https://wiki.openstreetmap.org/wiki/Key:tracktype) einbeziehen.
Wir müssen auch den vmax-Filter auf z.B. 30 km/h senken.

Geändertes Profil, Zeile 66: [car-eco-cruiser.brf.zip](https://github.com/devemux86/cruiser/files/11575147/car-eco-cruiser.brf.zip)
Ich kann diese Änderung in das nächste Cruiser-Update aufnehmen.

<img src="https://github.com/devemux86/cruiser/assets/3484020/ee4d1258-55cf-4fb9-82d6-059e0acdf454" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/17fc654c-a01d-4543-8ed0-792fe5fbff9e" width="50%"/>

**Reply by 0709wiwiwi** [reply]
If  you move shaping point 3 further onto the track no need to set 30kmh

**Reply by 1maxi2** [reply]
![Screenshot_20230527_094537_Cruiser](https://github.com/devemux86/cruiser/assets/124886589/d5996995-77a5-4cdf-8957-308d96ad0c76)
Welche Einstellungen hast du verwendet. Bei mir routet er nicht über die Brücke egal wie ich Punkt 3 verschiebe. ignore Barriere ist aktiviert. Geschwindigkeit ist auf 30 km/h eingestellt.

**Reply by devemux86** [reply]
@1maxi2 

Now you can use the straight line to pass the bridge.
See the route here: https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6013319

Next Cruiser will have updated BRouter profile with option "use_tracks".

---

Jetzt können Sie die gerade Linie benutzen, um die Brücke zu passieren.
Sehen Sie die Route hier: https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6013319

Der nächste Cruiser wird ein aktualisiertes BRouter-Profil mit der Option "use_tracks" haben.

**Reply by 1maxi2** [reply]
👍😁

**Comment by devemux86** [comment]
Even better we can make it optional:

[car-eco-cruiser-2.brf.zip](https://github.com/devemux86/cruiser/files/11575754/car-eco-cruiser-2.brf.zip)

<img src="https://github.com/devemux86/cruiser/assets/3484020/44666bc2-c537-4777-93e0-ca04179df212" width="50%"/>

**Reply by 0709wiwiwi** [reply]
Enduro ? 
https://wiki.openstreetmap.org/wiki/Key:tracktype

How far are you ultimately willing to go to get a result?
A track limit to set must sure still be  possible.
Or  you end up on the profile reroute.brf ( no rules).

**Reply by devemux86** [reply]
BRouter car profiles already have:
```
assign isgoodforcars =
                  ...
                  else if ( and highway=track tracktype=grade1 ) then true
```

Only this combination is enabled with the setting.
OpenStreetMap mentions for `tracktype=grade1`: "Solid. Usually a paved surface".

**Reply by 1maxi2** [reply]
> Noch besser: Wir können es optional machen:
> 
> [car-eco-cruiser-2.brf.zip](https://github.com/devemux86/cruiser/files/11575754/car-eco-cruiser-2.brf.zip)
> 
> <img alt="" width="50%" src="https://user-images.githubusercontent.com/3484020/241233530-44666bc2-c537-4777-93e0-ca04179df212.jpg">

Ich denke das wäre eine gute Lösung

**Reply by 1maxi2** [reply]
Eine Möglichkeit wäre wenn ich für diesen Abschnitt direkte Verbindungslinie verwende. Nicht ganz so schön aber funktioniert.
Über die Fähre läßt sich bei mir auch nicht planen obwohl Fähren zugelassen und 30km/h eingestellt 🤔

**Reply by devemux86** [reply]
@1maxi2 

BRouter now supports straight line for route segments.
In above [images](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011300) the ferry is used, but avoid tolls must be disabled.

BRouter unterstützt jetzt gerade Linien für Streckenabschnitte.
In den obigen [Bildern](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011300) wird die Fähre verwendet, aber die Mautvermeidung muss deaktiviert werden.

**Reply by 0709wiwiwi** [reply]
In thé first example before the change I did try thé straight solution, but for some unknown reason I could not set a beeline. Not able now to currently have more extra tests.

**Reply by 1maxi2** [reply]
> @1maxi2
> 
> BRouter unterstützt jetzt gerade Linien für Routensegmente. In den obigen [Bildern](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011300) wird die Fähre verwendet, die Vermeidung von Mautgebühren muss jedoch deaktiviert werden.
> 
> BRouter unterstützt jetzt gerade Linien für Streckenabschnitte. In den obigen [Bildern](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011300) wird die Fähre verwendet, aber die Mautvermeidung muss deaktiviert werden.

Sorry das war ein Fehler von mir. Hatte die Mautvermeidung aktiviert da ich normalerweise Autobahnen 
( = Maut in Österreich ) vermeide.
Mit deaktiviert Mautvermeidung kann man normal über die Fähre routen. Ein typischer Anwenderfehler von mir !

**Reply by WolfgangZ42** [reply]
[beeline.zip](https://github.com/devemux86/cruiser/files/11576845/beeline.zip)

**Reply by 0709wiwiwi** [reply]
https://youtu.be/4Ug5Io8A90o
By the original Cruiser default installed profiles in version20 > motorbiking  = car eco cruiser.brf
Solving the passage with a single beeline finally succeeded me now.
https://youtu.be/2ZnYVSa5g0Q

**Reply by devemux86** [reply]
@0709wiwiwi 

It is not only the bridge (tracktype=grade1), that can be passed with the new avoid_tracks or the straight line.

To continue the route on the north, you need to:
- Either move the waypoint 3 more to the north (as you correctly mentioned [above](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011984))
- Or lower the vmax filter to e.g. 30 km/h (see images [above](https://github.com/devemux86/cruiser/discussions/180#discussioncomment-6011300))

**Comment by 0709wiwiwi** [comment]
I had tested the Beeline function mainly by start to end track construction.

With this import track, however, all planner points with start and end were already placed
Emux had reported the bridge problem, a quick beeline solution was obvious.
Mistake: A too quick fix was expected
As there is not only that first bridge traverse problem but also an identical track routing problem further on.
![image](https://github.com/devemux86/cruiser/assets/54167201/41a4339a-cb5d-465c-a202-77348c0ffc9b)
A single beeline solution has to run over a **larger distance** than just crossing the first bridge. 

"Given the many pre-tests, I should have known this but still got surprised"

**Beeline or Straight Line.**
The start and end position of a Beeline is ALWAYS snapped or aligned onto a routable track section.
This is because the routable track sections MUST be connected with the Beeline section.  
You however can still divide a Beeline in parts by freely placed shaping point positions. 

**The import track that did not route by the version 20 original brf.**
The unsnapped shaping point 3 position is not decisive but the snapped position is. 
Using the function place waypoints on the street, this becomes visually much clearer. 
Above all, you need to look carefully where exactly the extra shaping point (4) is snapped then.

See the next gpx navigation track designs using the original version 20 brf files. 
[One beeline by version 20_car eco cruiser brf .zip](https://github.com/devemux86/cruiser/files/11583698/One.beeline.by.version.20_car.eco.cruiser.brf.zip)
[Two beelines by version 20 car eco cruiser brf.zip](https://github.com/devemux86/cruiser/files/11583699/Two.beelines.by.version.20.car.eco.cruiser.brf.zip)

In a navtrack, both shaping and via points are automatically snapped.

**Comment by devemux86** [comment, marked as answer]
BRouter routing on `highway=track` should work in [Cruiser 3.0.21](https://play.google.com/store/apps/details?id=gr.talent.cruiser) with its `car-cruiser` profile for Car / Motorcycle.

Activate the "Settings | Routing | BRouter (offline) | Car, Motorcycle | use_tracks" setting.

**Reply by 1maxi2** [reply]
Eben meine " Problemstrecke " mit der neuen Funktion " use track ausprobiert. 
Funktioniert einwandfrei ! Vielen Dank
Für die Umsetzung 😊
<img src="https://github.com/devemux86/cruiser/assets/124886589/1a679650-74b8-4d50-a4a9-b3ab1d1a8cfd" width="50%"/>

**Reply by DolomitesTom** [reply]
What is the function of highway=track?

**Reply by devemux86** [reply]
> What is the function of highway=track?

BRouter's default profile do not route on "highway=track" roads.
Please see the OpenStreetMap Wiki for the types: https://wiki.openstreetmap.org/wiki/Key:tracktype

With the setting "use_tracks" BRouter can use these roads via Cruiser's modified BRouter profile.

**Reply by DolomitesTom** [reply]
Ok understand thx... only something for the Africa Twin 🙂

**Reply by DolomitesTom** [reply]
The translation "Fahrbahnen benutzen" is not that good then. I would name it "Schotterwege benutzen" oder "Offroad Wege benutzen"

**Reply by devemux86** [reply]
> The translation "Fahrbahnen benutzen" is not that good then. I would name it "Schotterwege benutzen" oder "Offroad Wege benutzen"

@DolomitesTom however the [tracktype=grade1](https://wiki.openstreetmap.org/wiki/Key:tracktype) mentions "Usually a paved surface"?

**Reply by DolomitesTom** [reply]
OK, Grade 1 on the picture also looked pretty bad... then maybe
"kleinste und Offroad Wege benutzen" ?

**Reply by devemux86** [reply]
It may depend on the tags and the area.

e.g. this [example](https://github.com/devemux86/cruiser/discussions/182#discussioncomment-6019245) is a road to a hotel and this [example](https://github.com/devemux86/cruiser/discussions/180#discussion-5235795) is a bridge (routable by Kurviger).
(both [tracktype=grade1](https://wiki.openstreetmap.org/wiki/Key:tracktype))

**Reply by DolomitesTom** [reply]
Ok understand, road to a hotel or restaurant is acceptable, but I sometimes have supersport riders with me. They jump in my back when I chase them over such paths.

**Reply by 0709wiwiwi** [reply]
tracktype=grade1 not 2 3 4 5

**Reply by devemux86** [reply]
> tracktype=grade1 not 2 3 4 5

Currently the "use_tracks" option enables all `tracktype`.
Maybe we need different implementation?

**Reply by devemux86** [reply]
@DolomitesTom thanks for the translation!

Let's use the "Use gravel roads" / "Schotterwege benutzen".

**Reply by Ciutnik** [reply]
> tracktype=grade1 not 2 3 4 5

Definitely yes! Anything but „1” is more like „unpaved”.

**Reply by devemux86** [reply]
> Definitely yes! Anything but „1” is more like „unpaved”.

Could we add a "Use unpaved roads" option for our off-road friends?
It was a frequent request in the past.

**Reply by 1maxi2** [reply]
Würde ich für eine gute Idee halten.

**Reply by Ciutnik** [reply]
> Could we add a "Use unpaved roads" option for our off-road friends?
It was a frequent request in the past.

Certainly yes. Although the smartest solution would be the ability to set the level of avoidance - here for unpaved roads, but also for highways, prime, ferries, etc. We know it and we like it 🙂

**Reply by devemux86** [reply]
> Although the smartest solution would be the ability to set the level of avoidance - here for unpaved roads, but also for highways, prime, ferries, etc.

I have already noted it to examine it! :slightly_smiling_face:

**Reply by 1maxi2** [reply]
Ich glaube das macht die Sache zu komplex. Die neue kurviger app versucht das gerade. Je nachdem in welcher Gegend geplant wird wird es unterschiedliche Ergebnisse geben. 
Außerdem wenn ich offroadstrecken fahren will dann werde ich dies auswählen. Zwischenstufen ist wie ein bisschen schwanger.
Das ist natürlich nur meine persönliche Meinung. 
Da es ja sicher die Möglichkeit geben wird nur ja oder nein zu wählen würde mich eine Möglichkeit Zwischenstufen zu wählen nicht stören. Bin auf die Meinung anderer gespannt.
